tests/test_connection.py

```diff
@@ -48,6 +57,9 @@ def test_sharing_connections(self):
 
         connect('mongoenginetest', alias='testdb2')
         actual_connection = get_connection('testdb2')
+
+        # horrible, but since PyMongo3+, connection are created asynchronously
+        sleep(0.1)
```

Comment:
You're right. However, 0.1 seconds is already a lot. If you have a better suggestion, I'm very open to it